nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -335,19 +353,24 @@ in
                   ;;
               esac
             done
-            ''] ++ (map (pool: ''
+          ''] ++ (map (pool: (importLib {
+            zpoolCmd = "zpool";
+            awkCmd = "awk";
+            inherit pool cfgZfs;
+          }) + ''
```

Comment:
Here you overwrite existing function if multiple pools are defined. Can you only define each function once?

Reply:
If multiple *root* pools are defined?

Well, I suppose it'd be cleaner to not map across it. I'm not sure why this is a map in the first place, but let's do it right.